plugins/Tools/TranslateTool/TranslateTool.py

```diff
@@ -206,28 +208,22 @@ def setLockPosition(self, value: bool) -> None:
         model movement on the build plate.
         :param value: The setting state.
         """
-        for selected_node in self._getSelectedObjectsWithoutSelectedAncestors():
-            selected_node.setSetting(SceneNodeSettings.LockPosition, str(value))
+        self.setSettingToSelection(SceneNodeSettings.LockPosition, str(value))
 
     def getLockPosition(self) -> Union[str, bool]:
-        total_size = Selection.getCount()
-        false_state_counter = 0
-        true_state_counter = 0
-        if not Selection.hasSelection():
-            return False
+        return self.getBoolSettingFromSelection(SceneNodeSettings.LockPosition, "False")
 
-        for selected_node in self._getSelectedObjectsWithoutSelectedAncestors():
-            if selected_node.getSetting(SceneNodeSettings.LockPosition, "False") != "False":
-                true_state_counter += 1
-            else:
-                false_state_counter += 1
+    def setAutoDropDown(self, value: bool) -> None:
+        """Set lock setting to the object. This setting will be used to prevent
 
-        if total_size == false_state_counter:  # No locked positions
-            return False
-        elif total_size == true_state_counter:  # All selected objects are locked
-            return True
-        else:
-            return "partially"  # At least one, but not all are locked
+        model movement on the build plate.
+        :param value: The setting state.
+        """
+        self.setSettingToSelection(SceneNodeSettings.AutoDropDown, str(value))
```

Comment:
I think that this should also forcefully trigger a "sceneChanged" event, so that if this is changed by the user that it is automatically re-calculated.

RIght now if you disable drop objects in preferences and drag an object up and then click it in the translate tool, it remains floating for a bit (until anything else in the scene is updated).